CMakeLists.txt

```diff
@@ -25,21 +25,27 @@ if(${ENABLE_EPOLL_RECEIVE})
   add_definitions("-DENABLE_EPOLL_RECEIVE")
 endif(${ENABLE_EPOLL_RECEIVE})
 
-option(ENABLE_DOUBLE_RCVBUF "Enable double size of RCVBUF" OFF)
-if(${ENABLE_DOUBLE_RCVBUF})
-  add_definitions("-DENABLE_DOUBLE_RCVBUF")
-endif(${ENABLE_DOUBLE_RCVBUF})
+option(ENABLE_MODIFY_RECVBUF "Enable modify size of RECVBUF" ON)
```

Comment:
## ⚠️ Warning ⚠️ This changes a kernel parameter :rofl: 

```cpp
#ifdef ENABLE_MODIFY_RECVBUF
  {
    uint32_t opt_val = input_param_.socket_recv_buf, before_set_val,after_set_val = 0;
    if(opt_val < 1024)
    {
      opt_val = 106496;
    }
    socklen_t opt_len = sizeof(uint32_t);
    getsockopt(fd, SOL_SOCKET, SO_RCVBUF, (char*)&before_set_val, &opt_len);
    setsockopt(fd, SOL_SOCKET, SO_RCVBUF, (char*)&opt_val, opt_len);
    getsockopt(fd, SOL_SOCKET, SO_RCVBUF, (char*)&after_set_val, &opt_len);
  }
#endif
```

 The problem is with the syscall to [`setsockopt`](https://linux.die.net/man/2/setsockopt).  this is literally equivalent to run

```
sudo sysctl -w net.core.rmem_max=16777216
```

 which we do already in https://github.com/botsandus/auto-docker-images/blob/64c34c6c36b28f033c6f9c5aba696b7c58a41523/auto-base-image/scripts/setup_container.sh#L64

 so this would override this by my understanding :thinking: so will disable this crap

@richardw347, @doisyg please help me confirming that it's a horrible idea to let this ON by default

Reply:
SO, we decided to take this
James Newman:
> An extra impact is that it disables the kernel's automatic tuning for that socket and fixes the value - for TCP, it's normally in the min/max range set by tcp_rmem.  I'd only expect system-wide problems with things like this if you're doing a huge number of sockets, but preventing the kernel from tuning smaller.  However I expect it's one (or a handful) of high throughput sockets where we might want to fix a high buffer to handle high throughput or bursts without waiting for tuning.
It looks as though there isn't a "don't set it if the config isn't provided" path, which would be nicer.  However it will only affect this socket and there's probably a reason to tune it if they provided one!